pkg/container/container.go

```diff
@@ -298,3 +309,72 @@ func (c *containerModule) Delete(ns string, id pkg.ContainerID) error {
 
 	return container.Delete(ctx)
 }
+
+func (c *containerModule) ensureNamespace(ctx context.Context, client *containerd.Client, namespace string) error {
+	service := client.NamespaceService()
+	namespaces, err := service.List(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range namespaces {
+		if ns == namespace {
+			return nil
+		}
+	}
+
+	return service.Create(ctx, namespace, nil)
+}
+
+func setResolvConf(root string) error {
```

Comment:
I don't get what this method is trying to do (i know it is trying to write the etc/resolv.conf with static content) but the check and the stats, i don't get. I mean why exactly didn't u just use `ioutil.WriteFile` method. ?

Reply:
Cause I don't want to overwrite the content if there is any. So here I'm writing only of the file doesn't exist or if it's empty

Reply:
Does docker do that? because to me this sound weird. A container should always have a valid dns related to it's runtime network, not build time network. So this sounds a bit weird to me.

Reply:
Docker mount host resolv.conf to container resolv.conf using volume mount (AFAIK)

Reply:
yes as far as I've seen. Docker does write a file on the host filesystem and mounts it inside the container when it starts.
Docker does the same for hosts hostname and hosts file.

I can also do it the same way if you prefer. It just that since we already have a copy on write layer with flist I though maybe this is easier.

But the containerd has indeed a set of methods to deal with such a things:
```go
// WithHostHostsFile bind-mounts the host's /etc/hosts into the container as readonly
func WithHostHostsFile(_ context.Context, _ Client, _ *containers.Container, s *Spec) error {
	s.Mounts = append(s.Mounts, specs.Mount{
		Destination: "/etc/hosts",
		Type:        "bind",
		Source:      "/etc/hosts",
		Options:     []string{"rbind", "ro"},
	})
	return nil
}

// WithHostResolvconf bind-mounts the host's /etc/resolv.conf into the container as readonly
func WithHostResolvconf(_ context.Context, _ Client, _ *containers.Container, s *Spec) error {
	s.Mounts = append(s.Mounts, specs.Mount{
		Destination: "/etc/resolv.conf",
		Type:        "bind",
		Source:      "/etc/resolv.conf",
		Options:     []string{"rbind", "ro"},
	})
	return nil
}

// WithHostLocaltime bind-mounts the host's /etc/localtime into the container as readonly
func WithHostLocaltime(_ context.Context, _ Client, _ *containers.Container, s *Spec) error {
	s.Mounts = append(s.Mounts, specs.Mount{
		Destination: "/etc/localtime",
		Type:        "bind",
		Source:      "/etc/localtime",
		Options:     []string{"rbind", "ro"},
	})
	return nil
}
```